### PR TITLE
Bug 1855665: Rebase to kube-0.19.0-rc.3

### PR DESCRIPTION
**Pull request by coreydaley**
There were no tests for the tolerations overrides so i added some.

### CODE REVIEW

**Comment by coreydaley** [comment]
/retest

FAIL: github.com/openshift/openshift-controller-manager/pkg/serviceaccounts/controllers TestClearSecretAndRecreate 45.02s
Test passes locally, so giving it another try before I dig deeper.

**Comment by coreydaley** [comment]
/retest

FAIL: github.com/openshift/openshift-controller-manager/pkg/serviceaccounts/controllers TestUpdateOldStyleSecretWithoutKey 45.01s
Passes locally, different than previous error, trying again.

File: pkg/apps/deployer/deployer_controller_test.go [hunk 1]
```diff
@@ -1187,6 +1187,8 @@ func TestMakeDeployerPod(t *testing.T) {
 				p.Spec.Overhead = nil
 				p.Spec.EphemeralContainers = nil
 				p.Spec.TopologySpreadConstraints = nil
+				// k8s 1.19
+				p.Spec.SetHostnameAsFQDN = nil
```
Comment (adambkaplan) [inline pkg/apps/deployer/deployer_controller_test.go:1191]: fyi @frobware

File: pkg/build/controller/build/overrides/overrides_test.go [hunk 2]
```diff
@@ -357,3 +358,129 @@ func TestBuildOverrideAnnotations(t *testing.T) {
 		}
 	}
 }
+
+func TestBuildOverrideTolerations(t *testing.T) {
+	tests := []struct {
+		buildTolerations    []corev1.Toleration
```
Comment (adambkaplan) [inline pkg/build/controller/build/overrides/overrides_test.go:364]: nit - for table-driven tests:

1. Include a `name string` field. 
2. Run inside a separate test run:

```
for _, tc := range tests {
  t.Run(tc.name, func(t *testing.T) {
      ....
  })
}
```

This makes it easy to identify test cases that fail, and it _may_ also let the tests run in parallel.

File: pkg/build/controller/build/overrides/overrides_test.go [hunk 3]
```diff
@@ -357,3 +358,129 @@ func TestBuildOverrideAnnotations(t *testing.T) {
...
+			buildTolerations:    nil,
+			overrideTolerations: nil,
+			expected:            nil,
+		},
+		{
+			buildTolerations: nil,
+			overrideTolerations: []corev1.Toleration{
+				{
+					Key:   "toleration1",
+					Value: "value1",
```
Comment (adambkaplan) [inline pkg/build/controller/build/overrides/overrides_test.go:378]: nit - may as well throw in the toleration effect (ex `NoSchedule`).

File: pkg/cmd/controller/standalone_apiserver.go [hunk 4]
```diff
@@ -58,7 +59,9 @@ func RunControllerServer(servingInfo configv1.HTTPServingInfo, kubeExternal clie
 	authz := newBypassAuthorizer(remoteAuthz, "/healthz", "/healthz/ready")
 	handler := apifilters.WithAuthorization(mux, authz, legacyscheme.Codecs)
 	// TODO need audiences
-	handler = apifilters.WithAuthentication(handler, authn, apifilters.Unauthorized(legacyscheme.Codecs, false), nil)
+
+	//handler = apifilters.WithAuthentication(handler, authn, apifilters.Unauthorized(legacyscheme.Codecs, false), nil)
+	handler = apifilters.WithAuthentication(handler, authn, apifilters.Unauthorized(serializer.WithoutConversionCodecFactory{CodecFactory: legacyscheme.Codecs}), nil)
```
Comment (adambkaplan) [inline pkg/cmd/controller/standalone_apiserver.go:64]: Outside the scope of this PR - do we need this? Do integration tests run a standalone apiserver any more?

File: pkg/apps/deployer/deployer_controller_test.go (hunk 1 above)
Comment (adambkaplan) [reply pkg/apps/deployer/deployer_controller_test.go:1191]: fyi @tnozicka

File: pkg/build/controller/build/overrides/overrides_test.go (hunk 2 above)
Comment (coreydaley) [reply pkg/build/controller/build/overrides/overrides_test.go:364]: updated in next push

File: pkg/build/controller/build/overrides/overrides_test.go (hunk 3 above)
Comment (coreydaley) [reply pkg/build/controller/build/overrides/overrides_test.go:378]: updated in next push

**Comment by coreydaley** [comment]
@adambkaplan comments addressed, ptal

**Comment by coreydaley** [comment]
@adambkaplan all tests are passing and comments have been addressed.

File: pkg/apps/deployer/deployer_controller_test.go (hunk 1 above)
Comment (tnozicka) [reply pkg/apps/deployer/deployer_controller_test.go:1191]: lgmt, thx

**Comment by coreydaley** [comment]
@adambkaplan bump

**Pull request merged**